realm/userposts.gno

```diff
@@ -47,21 +58,120 @@ func (userPosts *UserPosts) AddThread(body string) *Post {
 	return thread
 }
 
+// If there is a UserPosts for followedAddr, then add it to following,
+// and add this user to its followers.
+// If there is no UserPosts for followedAddr, then do nothing. (We don't expect
+// this because this is usually called by clicking on the display page of followedAddr.)
+func (userPosts *UserPosts) Follow(followedAddr std.Address) {
+	followedUserPosts := getUserPosts(followedAddr)
+	if followedUserPosts != nil {
```

Comment:
Is it possible that the `followedUser` haven't any posts yet, but I still want to follow him anyway in case where he starts to post something?

Reply:
If `getUserPosts` returns nil, it means that the user has not made any interaction with GnoSocial. (The `followedAddr` would just be the address of some random account unknown to GnoSocial.) As I mentioned in the comment on line 64, `Follow` is usually called by clicking on the "Follow" button of the user's GnoSocial page. In this case `getUserPosts` will not return nil.